columnflow/tasks/ml.py

```diff
@@ -673,3 +683,219 @@ def merge(self, inputs, output):
+    ProcessPlotSettingMixin,
+    CategoriesMixin,
+    MLModelMixin,
+    ProducersMixin,
+    SelectorStepsMixin,
+    CalibratorsMixin,
+    law.LocalWorkflow,
+    RemoteWorkflow,
+):
+    """A base class, used for the implementation of the ML plotting tasks. This class implements
+    a `plot_function` parameter for choosing a desired plotting function and a `prepare_inputs` method,
+    that returns a dict with the chosen events.
+
+    Raises:
+        NotImplementedError: This error is raised if a givin dataset contains more than one process.
+        ValueError: This error is raised if `plot_sub_processes` is used without providing the
+        `process_ids` column in the data
+    """
+    sandbox = dev_sandbox("bash::$CF_BASE/sandboxes/venv_columnar.sh")
+
+    plot_function = PlotBase.plot_function.copy(
+        default="columnflow.plotting.plot_ml_evaluation.plot_ml_evaluation",
+        add_default_to_description=True,
+        description="The full path of the desired plot function, that is to be called on the inputs."
+        "The full path should be givin using the dot notation",
+    )
+
+    skip_processes = law.CSVParameter(
+        default=("",),
+        description="names of processes to skip; These processes will not be displayed int he plot."
+        "config; default: ('*',)",
+        brace_expand=True,
+    )
+
+    plot_sub_processes = luigi.BoolParameter(
+        default=False,
+        significant=False,
+        description="when True, each process is divided into the different subprocesses"
+        "which will be used as classes for the plot; default: False",
+    )
+
+    skip_uncertainties = luigi.BoolParameter(
+        default=False,
+        significant=False,
+        description="when True, uncertainties are not displayed in the table; default: False",
+    )
+
+    # upstream requirements
+    reqs = Requirements(
+        RemoteWorkflow.reqs,
+        MergeMLEvaluation=MergeMLEvaluation,
+    )
+
+    def store_parts(self):
+        parts = super().store_parts()
+        parts.insert_before("version", "plot", f"datasets_{self.datasets_repr}")
+        return parts
+
+    def create_branch_map(self):
+        return [
+            DotDict({"category": cat_name})
+            for cat_name in sorted(self.categories)
+        ]
+
+    def requires(self):
+        return {
+            d: self.reqs.MergeMLEvaluation.req(
+                self,
+                dataset=d,
+                branch=-1,
+                _exclude={"branches"},
+            )
+            for d in self.datasets
+        }
+
+    def workflow_requires(self, only_super: bool = False):
+        reqs = super().workflow_requires()
+        if only_super:
+            return reqs
+
+        reqs["merged_ml_evaluation"] = self.requires_from_branch()
+
+        return reqs
+
+    def output(self):
+        b = self.branch_data
+        return self.target(f"plot__proc_{self.processes_repr}__cat_{b.category}{self.plot_suffix}.pdf")
```

Comment:
consider using `get_plot_names` to get a list of output names with all requested file types.